File: c++/easy/climbingstairs.cpp
#include <iostream>

class Solution {
public:
    int climbStairs(int n) {
        if(n == 0 || n == 1) {
            return n;
        } 
        int prev = 1, curr = 1;
        for(int i = 2; i <= n; i++) {
            int temp = curr;
            curr = prev + curr;
            prev = temp;
        }
        return curr;
    }
};

File: python/easy/climbingstairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n is 0 or n is 1:
            return n

        prev = 1
        curr = 1
        for i in range(2, n + 1):
            temp = curr
            curr = prev + curr
            prev = temp

        return curr